### PR TITLE
implement select chain and account

### PR DESCRIPTION
**Pull request by rezaerami**
- add select chain dropdown
- add select account dropdown
- render existing chains and accounts based on wallet connect accounts

image
image
image
image

### CODE REVIEW

**Comment by sanderlooijenga** [comment]
Should we rename `Select Chain` to `Select Network`?

File: packages/apps/tools/src/components/Common/Layout/index.tsx [hunk 1]
```diff
@@ -31,13 +50,91 @@ export const Layout: FC<IProps> = ({ children }: IProps) => {
     },
   ];
 
+  const getNetworks = (): Network[] => {
```
Comment (sanderlooijenga) [inline packages/apps/tools/src/components/Common/Layout/index.tsx:53]: This logic is quite specific and belongs to the connect wallet functionality, right? If so; wouldn't it maybe be better to extract this logic to a better location?

File: packages/apps/tools/src/components/Common/Layout/index.tsx [hunk 2]
```diff
@@ -31,13 +50,91 @@ export const Layout: FC<IProps> = ({ children }: IProps) => {
...
+          const [, network] = account.split(':');
+          return network;
+        })
+        ?.filter((network) => network !== 'development'),
+    );
+
+    return Array.from(result ?? []) as Network[];
+  };
+
+  const getAccounts = (): string[] => {
```
Comment (sanderlooijenga) [inline packages/apps/tools/src/components/Common/Layout/index.tsx:66]: See comment above

File: packages/apps/tools/src/context/connect-wallet-context.tsx [hunk 3]
```diff
@@ -72,14 +109,15 @@ export const WalletConnectClientContextProvider: FC<
   );
 
   const connect = useCallback(
-    async (pairing?: { topic: string }) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async (pairing: any) => {
```
Comment (sanderlooijenga) [inline packages/apps/tools/src/context/connect-wallet-context.tsx:113]: There is no way of typing this parameter?

File: packages/apps/tools/src/components/Common/Layout/index.tsx (hunk 1 above)
Comment (rezaerami) [reply packages/apps/tools/src/components/Common/Layout/index.tsx:47]: not really, here we just want to get accounts that belong to a chain, wallet connect gives you all of the accounts with network as its prefix, so we want to visually separate them not logically

File: packages/apps/tools/src/components/Common/Layout/index.tsx (hunk 2 above)
Comment (rezaerami) [reply packages/apps/tools/src/components/Common/Layout/index.tsx]: same for accounts, this is only for separate accounts of a network visually

File: packages/apps/tools/src/context/connect-wallet-context.tsx (hunk 3 above)
Comment (rezaerami) [reply packages/apps/tools/src/context/connect-wallet-context.tsx:113]: not really, tbh i tried everything i knew and everytime it was giving me a different error

**Review by MRVDH — APPROVED** [review]
Checked the functionality etc and looks good. Toggle works well. The background colour of the select box changes when switching the theme still though, but let's keep that out for now until we have added the new Header component.

File: packages/apps/tools/src/components/Global/AccountNameField/index.tsx [hunk 4]
```diff
@@ -1,41 +1,135 @@
...
+          if (account.match(regex)) return account;
+        })
+        ?.map((account) => {
+          const [, , accountName] = account.split(':');
+          return accountName;
+        }),
+    );
+
+    return Array.from(result) ?? [];
+  };
```
Comment (alber70g) [inline packages/apps/tools/src/components/Global/AccountNameField/index.tsx:75]: - you can use `"".includes()` (no need to use regex for that, esp when you have to ignore a lint rule)
- you can return boolean in a filter lambda (no need to re-return when you're in a filter)
- you can pick the 3rd element straight from the split action
- A `Array.from(<SET>)` always returns an array, even if the set is empty

```suggestion
  const getAccounts = (): string[] => {
    return Array.from(new Set(
      accounts
        ?.filter((account) => account.includes(`kadena\:${selectedNetwork}`))
        ?.map((account) => account.split(':')[3]),
    ));
  };
```

**Review by alber70g — APPROVED** [review]
Minor comment

File: packages/apps/tools/src/components/Global/AccountNameField/index.tsx (hunk 4 above)
Comment (rezaerami) [reply packages/apps/tools/src/components/Global/AccountNameField/index.tsx:75]: valid point, all fixed

**Review by timoheddes — APPROVED** [review]

**Pull request merged**